File: 240403/hashmap 기본/hashmap-basic.py
N = int(input())

cmds = [list(input().split()) for _ in range(N)]

hashmap = {}

for i in range(N):
    cmd = cmds[i][0]
    key = cmds[i][1]

    if cmd == 'add':
        value = cmds[i][2]
        hashmap[key] = value
    
    if cmd == 'find':
        if key in hashmap:
            print(hashmap[key])
        
        else:
            print('None')
    
    if cmd == 'remove':
        hashmap.pop(key)

File: 240821/hashmap 기본/hashmap-basic.java
import java.util.*;

public class Main {

    public static int N;

    public static HashMap<Integer, Integer> M = new HashMap<>();

    public static void main(String[] args) {
        // 여기에 코드를 작성해주세요.

        Scanner sc = new Scanner(System.in);

        N = sc.nextInt();

        for (int i = 0; i < N; i++) {
            String cmd = sc.next();
            if (cmd.equals("add")) {
                int key = sc.nextInt();
                int value = sc.nextInt();
                M.put(key, value);
            }
            else if (cmd.equals("find")) {
                int key = sc.nextInt();
                if (M.containsKey(key)) {
                    System.out.println(M.get(key));
                }
                else {
                    System.out.println("None");
                }
            }
            else if (cmd.equals("remove")) {
                int key = sc.nextInt();
                M.remove(key);
            }
        }
    }
}